clientv3/maintenance.go

```diff
@@ -81,6 +81,7 @@ func NewMaintenance(c *Client) Maintenance {
 		dial: func(endpoint string) (pb.MaintenanceClient, func(), error) {
 			conn, err := c.Dial(endpoint)
 			if err != nil {
+				conn.Close()
```

Comment:
When `err` isn't nil, then `conn` should be nil. So it will run into nil pointer panic. 

It seems that 3.4 has different implementation, and we don't need to backport the PR to 3.4.